wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,344 @@
+            e_obj.add_field(name="Notice", value=f"**You are PERMANENTLY BANNED from\n{self.bot.guilds[0]}\n\n"
+                            "You may NOT rejoin the guild!**")
+            e_obj.set_footer(icon_url=self.bot.guilds[0].icon_url, text=self.bot.guilds[0])
+
+            await member.send(embed=e_obj)
+            await member.kick(reason="Not allowed back on server.")
+
+    @commands.Cog.listener(name='on_member_ban')
+    async def intercept(self, guild: discord.Guild, member: Union[discord.User, discord.Member]):
+        """Watches for a guild ban. The guild ban is undone and the user is banned via this ban system"""
+
+        # need to read the audit log to grab mod, date, and reason
+        logger.info(f"[Ban intercept()] guild ban detected and intercepted for user='{member}'")
+        try:
+            audit_ban = await self.bot.guilds[0].audit_logs(action=discord.AuditLogAction.ban).flatten()
+        except Exception as e:
+            logger.info(f'error while fetching ban data: {e}')
+            await self.mod_channel.send(f"Encountered following error while intercepting a ban: {e}\n" +
+                                        "**Most likely need view audit log perms.**")
+            return
+        audit_ban = audit_ban[0]
+        logger.info(f"[Ban intercept()] audit log data retrieved for intercepted ban: {audit_ban}")
+
+        # name, id, mod, mod id, date, reason
+        username = member.name + '#' + member.discriminator
+        mod = audit_ban.user.name + '#' + audit_ban.user.discriminator
+        mod_id = audit_ban.user.id
+        date = audit_ban.created_at
+        reason = audit_ban.reason if audit_ban.reason else 'No Reason Given!'
+
+        # update ban_list and db
+        self.ban_list.append(member.id)
+        await BannedUsers.insert_ban(username, member.id)
+        await BanRecords.insert_record(username, member.id, mod, mod_id, date, reason)
+
+        # unban
+        await self.bot.guilds[0].unban(member)
+        logger.info(f"[Ban intercept()] ban for {username} moved into db and guild ban was removed")
+
+        # report to council
+        e_obj = discord.Embed(title="Ban Hammer Deployed",
+                              colour=discord.Color.red())
+
+        e_obj.add_field(name="Banned User", value=f"**{username}**", inline=True)
+        e_obj.add_field(name="Moderator", value=f"**{mod}**", inline=True)
+        e_obj.add_field(name="Reason", value=f"```{reason}```", inline=False)
+        e_obj.add_field(name="Notification DM", value="NOT SENT, DUE TO NO COMMON GUILD\n", inline=False)
+        e_obj.set_footer(text="Intercepted Moderator Action")
+
+        if e_obj:
+            e_obj.timestamp = date.astimezone(pytz.timezone('Canada/Pacific'))
+            await self.mod_channel.send(embed=e_obj)
+        logger.info(f"[Ban ban()] Message sent to mod channel,{self.mod_channel}, of the ban for {username}.")
+
+    @commands.command()
+    async def initban(self, ctx):
```

Comment:
if this is a function/command that only needs to be run once to convert all the guild bans into wall_e bans, why not make it called from the constructor instead of making if a command instead?

Reply:
but then it would be called every time the bot restarts.

Reply:
is that bad?

Reply:
I see. so after this PR, you will make another immediately after removing this command?